src/core/installAddon.js

```diff
@@ -563,8 +572,26 @@ export function withInstallHelpers({
       WrappedComponent,
     )})`;
 
+    const mapStateToProps = (
+      state: AmoAppState | DiscoAppState,
+      ownProps: WithInstallHelpersProps,
+    ) => {
+      const { addon } = ownProps;
+      const currentVersion =
+        addon && addon.currentVersionId
```

Comment:
Why do you need the `addon &&` part? An addon must be provided. It is enforced twice, once with the Flow type (being `AddonType` only) and with an `invariant` call.

Reply:
If `addon &&` is required for some reasons, there is a bug somewhere and we should treat `addon` as a nullable. If it is only because of the two tests case that are failing in `test_installAddon.js`, then I guess we should fix them (and we can remove the spread of `addon` there).

Reply:
There's a problem somewhere because it's not just the tests that fail without this. It looks like the Flow definition at https://github.com/mozilla/addons-frontend/blob/2018.11.08/src/core/installAddon.js#L218 is incorrect, as I'm getting a `null` for `addon` when running the code. That may be a bug introduced by my patch - I'll dig deeper. It doesn't help that `src/amo/pages/Addon/index.js` isn't covered by Flow.

Reply:
>  It doesn't help that `src/amo/pages/Addon/index.js` isn't covered by Flow.

yep :(

I can see why the Flow type is incorrect: we don't have an add-on when we load an add-on detail page.

Reply:
As discussed on IRC leaving the `addon &&` for now is probably the best solution given our current code.